main.go

```diff
@@ -140,6 +145,36 @@ func init() {
 		zlog.Fatalf("--connections-per-host must be in the range [0,50]")
 	}
 
+	// If a source IP is specified, verify that the host is advertising the IP
+	// and has a chance to receive the responses.
+	if sourceIP != "" {
+		config.LocalAddressSet = false
+		providedAddr := net.ParseIP(sourceIP)
```

Comment:
Good catch, I just noticed that my errcheck addon was turned off. I will update that shortly.